src/components/text-editor/text-editor.tsx

```diff
@@ -34,6 +38,12 @@ import { buildDefaultMenu } from './menu/default-menu';
     styleUrl: 'text-editor.scss',
 })
 export class TextEditor {
+    /**
+     * Additional menu items to be displayed in the editor toolbar
+     */
+    @Prop()
+    public menuItems: EditorPluginButton[];
```

Comment:
Is it possible to remove (or hide) the default buttons?

If the "default" buttons are always there, and this property can only be used to _add_ buttons, I'm worried that we will be painting ourselves into a corner by doing that.

Would it be possible to expose the default set of buttons as an array that can be imported by the consumer? I'm thinking that the consumer could do something like the below if they want to change what buttons to show:

```ts
import { DefaultTextEditorMenuButtons, AllTextEditorMenuButtons } from '@limetech/lime-elements';

const menuWithExtraButtons = [ 
    …DefaultTextEditorMenuButtons,
    AllTextEditorMenuButtons['lift out'],
];

const menuWithDefaultButtonRemoved = [ …DefaultTextEditorMenuButtons ];
delete menuWithDefaultButtonRemoved.link;
```

Thoughts? Hot or not? 😅

Reply:
When we discussed this with Kia the path taken at that time was that we wouldn't allow the users to remove default buttons, but maybe this should be discussed further?

Reply:
I'm open for discussion, but I feel it's very strange if a consumer needs a rich text editor that supports everything and offers all the shortcuts and keybindings, but they want to hide the basics of a text editor (such as **B**, _I_, H, lists, etc…).

I can empathize if someone dons't want to "remind" the end-users that:

> Hey by the way! You can have code-blocks and horizontal rules, etc… Feel free to ruin the text you're writing here!

and I can also empathize with someone who might need to remind their end-users that:

> Hey, it's ok to write code or code-blocks in this context, and we have the text editor with those features for you.

But I can't empathize with others who want to hide the basics. Like:

> Hey! You can't make anything bold. But italics is Ok. And no lists are allowed!

And in reality they can't. Since the keybindings and markdown short commands are working no matter if the button is shown or not.

So what do you say? 😊

Reply:
Well, I see plenty of good reasons why we shouldn't, so if the best argument for why we should is "why not", then let's not please.

Reply:
We can have a huddle if you need my reasoning explained 😊

Reply:
I didn't really understand if you agreed or disagreed with me 😅

Reply:
Haha! Sorry, I was feeling a little stressed. Too much to do 😅 Shouldn't allow myself to feel stressed though. Stupid Adrian! 🙈

Anyway: I disagree with you! 😄

Reply:
OK 😄 is it a blocker or what?

Reply:
I think it is. What we go with here, we will likely be stuck with. So we need to choose something that's flexible. If we feel we need prevent the consumer from hiding certain buttons, we can do that in a way that doesn't effectively hard-code what buttons are the default set, and make it virtually impossible for us to ever change that.

Reply:
Can we deal with this in a follow up PR? The component are private as of now. It would be good to get stuff in so that we can start iterating

Reply:
All the other things, yes. This one, hrmph… 😣 I think that if this is merged, even while private, it won't ever be changed. There will be 5 new PRs opened, which all build on the current implementation, and then add-ons will start building on it too, and it can never be fixed. And I'll be the one that has to deal with it ten years down the line when you all have left for other jobs… 😞

Reply:
Don't get me wrong, I'd love to implement it myself right away, but I've already got my plate full. I need to get a backport done today, and then there's some problem with the automatic update of lime-crm-components that Bef reported, and then I have the stuff I need to do in my own part of the cycle.

This really shouldn't be a difficult thing to remake. The straightforward way ought to be easier to implement than what's already there.

Reply:
(if this helps) Personally, I'm completely fine if we don't even allow anyone to add back what we excluded from ProsMirror's default menu to start with. I believe this is a flexibility that "nobody" really needs. Our type of use-case  requires only the most standard rich-text editing things.

Reply:
I see your point here, we still need this functionality to strip away some default menu items from prosemirrors basic menu. What do you think about this suggested compromise for now so we can move on? https://github.com/Lundalogik/lime-elements/pull/2905#discussion_r1574642912

Reply:
The question of whether the consumer should be able to change things is, frankly, almost completely irrelevant for my argument. My argument is that _we_ have to be able to change our mind. The current solution paints us into a corner that will be an absolutely _massive_ headache for us if we so much as decide that "actually, sometimes maybe we would prefer not to show the link-button".

If you can't see that, then _please_ trust me because I _can_ 🙏